src/wallet/wallet.cpp

```diff
@@ -3950,6 +3957,7 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
         {
             CWalletDB walletdb(*walletInstance->dbw);
 
+            LOCK(walletInstance->cs_wallet);
```

Comment:
This should be before the line above. Anyway, there is no need to lock this wallet since it's a fresh instance, unknown to the remaining system. Unless you are adding the lock because of other asserts and if so please commit them to justify this change?

Reply:
Now moved before the line above. Without that lock we'll trigger Clang thread safety analysis warnings when #11634 is merged due to:
```
src/wallet/wallet.h:    std::map<uint256, CWalletTx> mapWallet GUARDED_BY(cs_wallet);
```

As agreed upon in https://github.com/bitcoin/bitcoin/pull/11226#issuecomment-341496939 all guard/lock annotations are added in #11634 and all extra locking is submitted as separate PR:s (such as this one).

The former are compile-time only whereas the latter might change run-time behaviour (and thus needs extra scrunity!).

Reply:
> As agreed upon in #11226 (comment) all guard/lock annotations are added in #11634 and all extra locking is submitted as separate PR:s (such as this one)

I think you swapped the PR numbers?

Reply:
Yes, you're right - the annotations are added in #11226 :-)